xml_analyzer/data_container.py

```diff
@@ -6,21 +6,31 @@ def __init__(self):
         self.mapping_element = collections.namedtuple('property_value_mapping', ['property', 'value'])
         self.mappings = dict()
         self.mappings['variables'] = dict()
+        self.mappings['autocalc'] = dict()
         self.result_property = ''
         self.dependencies = list()
         self.constraint_specification = ''
 
     def add_prop_val_mapping(self, prop: str, value: str):
-        map_element = self.mapping_element(prop, value)
-        self.mappings['variables'][prop] = value
+        self.mappings['variables'][prop] = float(value)
 
     def add_result_property(self, prop):
         self.result_property = prop
 
+    def get_result_property(self) -> str:
+        return self.result_property
+
+    def add_dependency_mapping(self, prop: str):
+        self.mappings['variables'][prop] = 'dep'
+
     def add_auto_calc_mapping(self, sysml_type_name: str, autocalc_method: str, values: list):
-        self.mappings['autocalc'] = dict()
+        try:
+            map_elements = self.mappings['autocalc'][autocalc_method]
+        except KeyError:
+            map_elements = list()
         map_element = self.mapping_element(sysml_type_name, values)
```

Comment:
maybe put the rest of the function into an else block. Exception handling in python consists of try - except - else -finally (else and finally are optional)

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/6614ef2d31e5e3284ee434bbd113e446107a9606